src/ucs/arch/generic/cpu.h

```diff
@@ -8,9 +8,15 @@
 #ifndef UCS_GENERIC_CPU_H_
 #define UCS_GENERIC_CPU_H_
 
+#include <ucs/type/status.h>
 #include <sys/time.h>
 #include <stdint.h>
 
+typedef struct ucs_cache_size {
+    size_t       cache_size;
+    ucs_status_t status;
+} ucs_cache_size_t;
```

Comment:
this struct should be defined only in C file, the fact taht we cache the cache size is implementation detail which should not be exposed in API
ucs_arch_get_cache_size() should return ucs_status_t

Reply:
@yosefe the problem is in reading of complete table at once: this table is used in general/FS cache reading and in arch-specific (CPUID for x86) call which is reading whole table too. add one more internal header?

Reply:
ok. bu do we really have to keep ucs_status_t? maybe just 0 (or -1) to mean "unknown"?

Reply:
actually looking how to eliminate this datatype... going to use just array of size_t where zero value means not supported cache

Reply:
> ucs_arch_get_cache_size() should return ucs_status_t

not sure why... but ok - will return status